src/library/simplified/extension/cesc-LT-channel-country-network-DTL.js

```diff
@@ -17,34 +20,44 @@ for(var i=0; i < length; i++){
     }
 }
 
-b["cescLastTouchTimeStamp"] = lastTouchTimeStamp;
-b["cescLastTouchMarketingCode"] = lastTouchMarketingCode;
+
 b["cescPriorTouchTimeStamp"] = priorTimeStamp;
 b["cescPriorTouchMarketingCode"] = priorMarketingCode;
 
-
-var dateDifference = parseInt((new Date() - new Date(lastTouchTimeStamp))/(1000*60*60*24));
-if(dateDifference <= 30 && dateDifference >= 0) {
-    b['cescLastTouch_isexpired'] = false;
-}else {
-    b['cescLastTouch_isexpired'] = true;
-}
-
-if(lastTouchMarketingCode.split(".").length >= 4){
-    b["cescChannel"] = lastTouchMarketingCode.split(".")[0];
-    b["cescCompany"] = lastTouchMarketingCode.split(".")[1];
-    b["cescCountry"] = lastTouchMarketingCode.split(".")[2];
-    b["cescNetwork"] = lastTouchMarketingCode.split(".")[3];
-}
-
-
 var fullMarketingCode = lastTouchMarketingCode;
 var patrialMarketingCode = b["context.marketingAttribution.campaignId"];
 
+b["cescDTL"] = "";
 if(fullMarketingCode != undefined && patrialMarketingCode != undefined && fullMarketingCode.indexOf(patrialMarketingCode) > -1 ){
     var marketingCodeAndDTL = fullMarketingCode.split(patrialMarketingCode + ".");
     if( marketingCodeAndDTL.length >= 2 ){
         b["cescDTL"]  = marketingCodeAndDTL[1];
-        b["cescLastTouchMarketingCode"] = patrialMarketingCode;     //This is to handle the case where there is DTL in the lastTouchMarketingCode.
     }
 }
+
+
+// Use CID for last touch
+
+b["cescLastTouchTimeStamp"] = b["context.marketingAttribution.campaignIdTimestamp"];
+b["cescLastTouchMarketingCode"] = b["context.marketingAttribution.campaignId"];
+var dateDifference = parseInt((new Date() - new Date(b["cescLastTouchTimeStamp"]))/(1000*60*60*24));
+
+b['cescLastTouch_isexpired'] = "";
+if(dateDifference <= 30 && dateDifference >= 0) {
+    b['cescLastTouch_isexpired'] = false;
+}else {
+    b['cescLastTouch_isexpired'] = true;
+}
+
+// Load rule looks for country and network, and if these are undefined in the udo the load rule will break and tag will not fire even it other conditions are met.
+// Therefore, initializing it to empty so load rule doesn't break.
+b["cescChannel"] = "";
+b["cescCompany"] = "";
+b["cescCountry"] = "";
+b["cescNetwork"] = "";
+if(b["cescLastTouchMarketingCode"].split(".").length >= 4){
+    b["cescChannel"] = b["cescLastTouchMarketingCode"].split(".")[0];
```

Comment:
Same as above, I didn't add this, it has been like this for a while, but I can update.